src/include/processor/operator/aggregate/aggregate_hash_table.h

```diff
@@ -9,6 +9,13 @@
 namespace kuzu {
 namespace processor {
 
+template<typename type>
+bool compareEntryWithKeys(const uint8_t* keyValue, const uint8_t* entry) {
```

Comment:
ditto try put back. We have many places that are written in a similar way. I don't know why some of them can be compiled while others we need to change

Reply:
Could you give an example of another place which didn't require changing? That might provide some insight into other ways the issue can be worked around.

It's worth noting that it seems to work fine as long as the function pointer isn't returned from another function (that's the point where the function resolution fails).

E.g. [this](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjEEgCspAAOqAqETgwe3r7%2ByanpAiFhkSwxcVyJdpgOGUIETMQEWT5%2BAVU1AnUNBEUR0bEJtvWNzTltwz2hfaUDFQCUtqhexMjsHAD06wDUACoAnkmYW3tLxFtoWFsIsZikWyRbtKhM6FuGW5iqrEn0AHQmGgAguYAMyhZDeS4mEFuKheBgdZi0aHYAHAoEQpgKBS7TAKAhbEwAdisQKSXiitDwyBAaK29K2BEwLB%2BTCZ0LcBAOjFYR0BAH0dgBNZSokGooEMrb4tnUrYAN1QeFeTPx/LhCMcAggAuFooVYi8mDmhJJdKlDPN9OJABE0bboaT0YDFcrGXiCBATcSnVL8egQCANYiOa7XhBQgQ5iiDd4jtCbYSzAA2HYewOqgjq%2BEhmGRlGOq2xo0QLjRkFO20cBa0TjxXh%2BDhaUioThuazWaWnVZJkE8UgETTVhYAa38iVrHEkDaHLc4vAUIA0A6HCzgsCQaBZdFi5EoW6SO7i8uQSSS/PlXAAnPyDJn%2Bapk9IsPLqZgAGp4TAAdwA8odG37GhaCZYhFwgKJZyiUIGj2Th%2B2g5hiD2X8om0apB24XgtzYQRfwYWg4KbXgsCiLxgDcMRaEXLDSCwFhDGAcRiLovBiAwvB5TxWdPmqLwmXg3hI0wSdmypKJiFgjwsFnAhiDwFhBIWKgDGABRPx/f9GEEmRBBEMR2CkXT5CUNRZ10Lh9EYlAO0sfQ8CiRdIAWVAki1BgaN4VAuOIeSsCcr1bBEjjnAgVxRj8SzgimEoyj0FI0nciL4vydzeli2YgvsdyuhGTwWj0docomdL%2BnKIZumSyyZUaUqZnKBYFG7QyazrGcWNbDgtkfSQthYBQTwVa9fjvD0tggXBCAeUEy14TCtDmUdx30Thp1IRtm06hclxXYi1xgRAQCWAhyQIPcIAPI9wl5ThVAADmTABaJ8tmAZBkC2LgQV%2BftMHwIg/L0fg9NEcQjOBkyVHUFiLNIb9JKSJSVo4et1tnTrf34077iobqXv6wbLyvEa2TGiAPG3egzhmuY5tXJazEkX4rxZ1m2bZ5NkbWjavPnWwdvm6tSHXQ7Lqp86xYGYAvssl83w0v8AJ04DQPAyCWMQ2CdM15DUPQhwdJwxgCHwwjZ1I8jKNoaidPoxjmObfB2JqLiaObXjkH4tZ%2B2E0TeHEyTkOktZmzkhSkZUpg1IVrTAN4CH9LB6QIcUKHzNyAwjBsyxrHsxz4BctyMk8lsfL87jnKykK/DChh3HynIorruq4sshKCkyBvIryRKMhbzKitqCYqqrxFcsmYoysK4eu%2Bn7p%2B4axZllWCRWpR9rNs4bqHue3q3o%2Br6fvGyaAd7WbdoWhmJ1WjfeY4bbl0FxbkbMW%2B53vi/h1IHy0mcSQgA%3D%3D) works, but [this](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjEEtIADqgKhE4MHt6%2B/qRJKY4CIWGRLDFxUraY9vkMQgRMxAQZPn4BdpgOabX1BIUR0bHxtnUNTVmtwz2hfSUDUgCUtqhexMjsHAD06wDUACoAngmYW3tLxFtoWFsIsZikWyRbtKhM6FuGW5iqrAn0AHQmGgAguYAMyhZDeS4mEFuKheBgdARiaHYAHAoEQpgKBS7TAKAhbEwAdisQISXiitDwyBAaK29K2BEwLB%2BTCZ0LcBAOjFYR0BAH0dgBNZSokGooEMrb4tnUrYAN1QeFeTPx/LhCOqEAFwtFCrEXkwc0JJLpUoZZvpxIAImibdDSejAfj0CAQBrEQwOYrlVsIKECHMUYy8QQIMbiY6pcRMARlgxCWYAGw7UNu1UEdXwz0cgMoh12onWjgLWicACsvD8HC0pFQnDc1ms0tOq0TIJ4pAImhLCwA1v5y/pOJIqz265xeAoQBouz2FnBYEg0Cy6LFyJQVwk13F5cgEgl%2BfKuABOfkGDP81RJ6RYeXUzAANTwmAA7gB5Q7Vzs0WhM4jThAUTjlEoT1HsnCdqBzDEHs75RNo7TdtwvArmwgjvgwtAQTWvBYFEXjAG4Yi0NOKGkFgLCGMA4i4RReAxh08p4uOnztF4TKQbwAaVOOVJRMQ4EeFg44EMQeAsFxCxUAYwAKM%2Bb6fowXEyIIIhiOw5T8IIigqOodG6Fw%2BjUSgTaWPoeBRNOkALKgCTVGRvCoMxxDiVg1nhhUVRpC4DDuJ4zR6MEUzFKUei5KkAijH4RkRdUvShbMXlIZ0EzRXobSel0DQJf0ZRDN06VGTKOUhXlEgLAoraaaWFZjnR9YcFs16SFsLAKHuCqnr8F6hn6uCEA8oJcHMvDIVocz9oOw4cKOpDVrWjVTjOc64QuMCICASwEOSBAbhAW47uEvKcKoAAcSYALQ3lswDIMgWxcCCvydpg%2BBEG5ejacIojiFpsi6Wo46GaQr6CQkUkzZW83jo174cbt9xUM1N3tZ1x4nj1bJ9RAHirvQZzDaNq0TVNZiSL8J5U9TNM00mM1zQtTmTrYK3jSWpCLpth0E/tPMDMAT1GXeD4KR%2BX4qb%2B/6AcBdHQeBKny7B8GIQ4KloYwBCYdh474YRxG0KRKmUdRtG1vgjGOMxZG1mxyAcWsnY8WWdH8YJsHCWstZiRJkMyUwcli0p368N96l/dI32A/pta6AEBhGKZljWBZVnwLZ9lpI5dYuW5LE2clnq%2Bf5mQxYEfm5TM%2BVxWkRU5MkkUMJXYXFZUKUCNljQBVkrfeR3EzN0lJVd6XGUD2VVcVYsyyrFPLvQ0zE5NedV03XdD1PS9/XvUNZgdsT7OTaQA5mEOLuM7DLPTrOh%2B1RwZj1YtLO36QLkpM4khAA) does not.

Reply:
I see. This is a good point. Then I wonder if we also pass functor reference rather than return a functor. 

For example in `vector_boolean_operation` we can do
```
void VectorBooleanOperations::bindExecFunction(
    ExpressionType expressionType, const binder::expression_vector& children, scalar_exec_func& func) {
    if (isExpressionBinary(expressionType)) {
        func = bindBinaryExecFunction(expressionType, children);
    } else {
        assert(isExpressionUnary(expressionType));
        func = bindUnaryExecFunction(expressionType, children);
    }
}
```
I think @acquamarin and @ray6080 should also comment on this. If so, @benjaminwinger you can ignore my other comment and do this in a separate PR.

Reply:
> I see. This is a good point. Then I wonder if we also pass functor reference rather than return a functor. Because expose static function in header files is not a good coding style
> 
> For example in `vector_boolean_operation` we can do
> 
> ```
> void VectorBooleanOperations::bindExecFunction(
>     ExpressionType expressionType, const binder::expression_vector& children, scalar_exec_func& func) {
>     if (isExpressionBinary(expressionType)) {
>         func = bindBinaryExecFunction(expressionType, children);
>     } else {
>         assert(isExpressionUnary(expressionType));
>         func = bindUnaryExecFunction(expressionType, children);
>     }
> }
> ```
> 
> I think @acquamarin, @ray6080 and @benjaminwinger should all comment on this. If so, @benjaminwinger you can ignore my other comment and do this in a separate PR.

Sounds good to me.